index.js

```diff
@@ -1,88 +1,121 @@
 
-// console.log(name)
+const name = "Hanna"
+console.log(name)
+
 // => "Raffy"
 // name = "Not Raffy" 
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
 // console.log("%cQuestion 3", "color: red") 
-
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
+function drinkWater(current_thirst_level){
+  console.log('Man I sure am thirsty')
+  current_thirst_level -=1
+  console.log("Ahh that hits the spot")
+  return current_thirst_level
+}
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
+console.log("%cQuestion 4", "color: red") 
+function sameSameButDifferent(num, maybeNum){
+  if(num === maybeNum){
+    return "same same"
+  } else if (num == maybeNum){
+      return "same same (but different)"
+  } else {
+    return "different"
+  }
+ 
+}
+
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade){
+    student.grade = grade 
+}
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
-// // => "Duane"
+console.log("%cQuestion 6", "color: red")
+
+function printNameAndPhones(user){
```

Comment:
Nice. I would consider renaming the anonymous function's parameter on line 94 to something that's a little more indicative of what it's going to actually represent. `user_hash`, aside from using Ruby naming conventions (remember hashes are objects in Javascript!) is actually misleading because it's going to represent each element of our `users` array. So a better name might be `user`! 

You're still accomplishing the same thing and this is ultimately correct, but try to be as intentional as you can with naming.